### ISSUE TITLE
CUDA TENSOR RuntimeERROR

### ISSUE THREAD

**Issue by ravikakaiya** [PyTorchLightning/metrics#1047]
## 🐛 Bug

While Passing a cuda tensor we encounter a RuntimeERROR : Expected all tensors to be on the same device, but found at least two devices, cuda:0 and cpu!

### To Reproduce

Run the following code.
#### Code sample
```py
from torchmetrics import PeakSignalNoiseRatio
import torch
preds = torch.rand([3, 128, 128], generator=torch.manual_seed(42))
target = preds * 0.75
preds = preds.cuda()
target =target.cuda()

psnr_loss = PeakSignalNoiseRatio()
avg_psnr = psnr_loss(preds, target)
print(avg_psnr)
```
### Expected behavior

Should return a tensor of a value. 

### Environment

- TorchMetrics version (and how you installed TM, e.g. `conda`, `pip`, build from source): TorchMetrics version 0.8.2
- Python & PyTorch Version (e.g., 1.0): python 3.9.12 , pytorch 1.11.0
- Any other relevant information such as OS (e.g., Linux):OS : Linux

**Comment by Borda** [comment]
it is expected, that you need to have also the metric on the sam device or use PL logging which does it for you... :rabbit: 
see: https://github.com/PyTorchLightning/metrics/issues/608#issuecomment-963153184

**Comment by ravikakaiya** [comment]
Thank You @Borda

**Comment by lsecerovic** [comment]
@Borda can you please comment how to use PL logging so that it does this for us?
